consul/consul.go

```diff
@@ -98,10 +98,36 @@ func (r *ConsulAdapter) buildCheck(service *bridge.Service) *consulapi.AgentServ
 	return check
 }
 
+func (r *ConsulAdapter) postAttributes(service *bridge.Service) error {
+	for k, v := range service.Attrs {
+		data := new(consulapi.KVPair)
+		data.Key = buildServiceKey(service, k)
+		data.Value = []byte(v)
+		log.Println("Updating Consul KV: ", data.Key, v)
+		_, err := r.client.KV().Put(data, nil)
+		if err != nil {
+			log.Println("Error Updating Attribute: ", err)
+		}
+	}
+	return nil
+}
+
+func buildServiceKey(service *bridge.Service, key string) string {
+	return fmt.Sprintf("service/%s/meta/%s", service.Name, key)
+}
+
 func (r *ConsulAdapter) Deregister(service *bridge.Service) error {
+	r.removeAttributes(service)
 	return r.client.Agent().ServiceDeregister(service.ID)
 }
 
+func (r *ConsulAdapter) removeAttributes(service *bridge.Service) error {
+	for k, _ := range service.Attrs {
+		r.client.KV().Delete(buildServiceKey(service, k), nil)
```

Comment:
@zarlant if you have multiple instances of a service with the same SERVICE_NAME, and one is deregistered, will this remove the K/V entry that may still be relevant to remaining instances of the service?